backend/routes/content.py

```
@@ -441,32 +577,38 @@ def complete_unit(unit_id, user_id):
 
+    # TODO: these events shouldn't need to be triggered if the unit is already marked as complete?
     # Trigger the UNIT_COMPLETION event to award badges
+    unit = Unit.query.filter_by(id=unit_id).first()
     badge_awarding_service = BadgeAwardingService(user_id)
     badge_awarding_service.check_and_award_badges(
-        EventType.UNIT_COMPLETION, user_unit=user_unit
+        EventType.UNIT_COMPLETION,
+        user_unit_completed=user_unit.completed,
+        unit_title=unit.title,
     )
+    logger.debug(f"Badges checked for unit {unit_id} completion")
```

Ruff (0.8.2)

Logging statement uses f-string

(G004)